### ISSUE TITLE
[Feature Request] Improve how we can control the expansion-panels items state

### ISSUE THREAD

**Issue by AlexandreBonneau** [vuetifyjs/vuetify#3271]
### New Functionality
Currently, there is no easy way to programmatically open or close all the `v-expansion-panel-content` at once.

It would be handy to have a prop on the `v-expansion-panel` parent element that we could use by passing the following values:
- `'open'` : would open all the `v-expansion-panel-content`
- `'close'` : would close all the `v-expansion-panel-content`
- `'toggle'` : would toggle each `v-expansion-panel-content` `isActive` states
- `'freeze'` : would prevent the user to modify the `isActive` state of all the `v-expansion-panel-content` child components

### Improvements
Currently you cannot 'freeze' the component, and there is no easy way (if any) to open/close/toggle them.

### Bugs or Edge Cases it Helps Avoid
None, it's a feature request.

**Comment by wadestuart** [comment]
toggle as simple as !isActive no?

**Comment by AlexandreBonneau** [comment]
Well, you still have to duplicate the `:value=variableThatControlTheOpenCloseState` on each `v-expansion-panel-content` component.

This is [how I toggle the open/close state globally](https://codepen.io/AnotherLinuxUser/pen/parROL) for now.
Notice all the duplicated ` :value="showAllForms"`.

**Comment by appurist** [comment]
I have been using the `:value` prop to control the panel expansion. I don't see a problem with this in its current form (if you have multiple panels and want them _all_ to be controlled by a single data item -- and I may be missing something but -- isn't adding `:value="showAllForms"` to each panel exactly what we'd want here?  I used an expression to control a single current panel in my codesandbox [here](https://codesandbox.io/s/l99jy01w79?module=%2Fexpander.vue), but that goes to show how flexible it is by having a per-panel `:value` prop that controls this.

**Comment by appurist** [comment]
I'd love to see higher-level options, i.e. move the expansion choices to the parent `v-expansion-panel` component rather than the `v-expansion-panel-content`.  In that case, there could be options like `force-open` (for accordion-like or all-panels behavior), and `single-expand` where `v-model` could indicate the current expanded panel (or perhaps an array of indices for multiple).  I see that as more `v-expansion-panel` options rather than the current `:value prop in `v-expansion-panel-content`.

**Comment by appurist** [comment]
I haven't tested with the updates in #3324 but that looks wonderful, and if the answer here now to `:input=` for this is just "don't do that", I'm comfortable with that. ;)  There is the mystery of why `:input` stops one side of the events, but the new approach means we probably don't even need them anymore, and we don't need `:input` now, and the events work with `:input` removed anyway.  From my perspective, you can close this one now, but that's up to you.  Kudos on the new `v-model` change. 👍

**Comment by AlexandreBonneau** [comment]
#3324 looks great @nekosaur :+1:

I like the `next` and `prev` features and the very precise control over which expansion panel is opened/closed when passing a simple array to the parent.

Perhaps I missed it, but I do not see any way to *'freeze'* the expansion panels.
My use case was to prevent the user to modify the open/close state of the whole expansion panels (ie. in an admin page), but if we want to be coherent with the `v-model` change, perhaps it would be even better if we could 'freeze' each panel separatedly.

Perhaps with an additional `:freeze` prop like so? (based on your PR example):
```html
<template>
  <v-app dark>
    <v-expansion-panel
        v-model="model"
        :expand="expand"
        :freeze="freeze"
        focusable
    >
      <v-expansion-panel-content v-for="item in items" :key="item">
        <div slot="header">Item</div>
        <v-card>
          <v-card-text>Lorem ipsum dolor sit amet, consectetur adipiscing elit.</v-card-text>
        </v-card>
      </v-expansion-panel-content>
    </v-expansion-panel>
    <!-- Modifying the model -->
    <v-btn @click="model = model - 1">prev</v-btn>
    <v-btn @click="model = model + 1">next</v-btn>
    <v-btn @click="model = [true,true,true,true,true]">all</v-btn>
    <v-btn @click="model = []">none (empty array)</v-btn>
    <v-btn @click="model = null">none (null)</v-btn>
    <!-- Expand actions -->
    <v-btn @click="expand = !expand">toggle expand</v-btn>
    <!-- Freeze actions -->
    <v-btn @click="freeze = freeze - 1">freeze previous</v-btn>
    <v-btn @click="freeze = freeze + 1">freeze next</v-btn>
    <v-btn @click="freeze = [true,false,false,true]">Freeze some (first and last) (array)</v-btn>
    <v-btn @click="freeze = [true,true,true,true]">Freeze all (array)</v-btn>
    <v-btn @click="freeze = true">Freeze all (true)</v-btn>
    <v-btn @click="freeze = 'all'">Freeze all ('all')</v-btn>
    <v-btn @click="freeze = []">Unfreeze all (empty array)</v-btn>
    <v-btn @click="freeze = null">Unfreeze all (null)</v-btn>
    <!-- What about toggling the freeze states? -->
  </v-app>
</template>
```

With this and if we want to provide the same behavior than for the `model`, toggling the `freeze` states would require an additional prop; not sure if you would be willing to add it too.

**Comment by AlexandreBonneau** [comment]
Also, perhaps the `expand` prop should be renamed as `toggle`?
Currently `!expand` means collapse to me, not toggle; am I missing something?

**Comment by nekosaur** [comment]
Yes, the expand name has been bothering me as well. Perhaps we can deprecate it in favor of something else. Although not sure if we're using the same name somewhere else for consistency.

I'm not sure about the freeze functionality. I could possibly see a usage for freezing the entire panel, using a boolean prop. What is the exact real scenario where you would need to individually control the freezing?

**Comment by AlexandreBonneau** [comment]
I'm using the expansion panel component almost as a vertical stepper (since with the stepper you can really only use 3 steps, any additional ones being hidden by default).

In each expansion panel content, I have a part of (a pretty big) form.
Once the user has entered the correct info in an 'expansion panel content form', then it freezes so it cannot be closed.

Another use case is that I switch between a simple and a detailed view;
The simple view is just a basic form, while the detailed one is the aforementioned expansion panel component form above.
When the user switch to the detailled view, the very first expansion panel content should *always* be displayed open, since it contains the default very basic data needed by the form (the other expansion panel content containing optional data).

In short, even if it's hard to describe, there are use cases for precisely controlling which panel can be frozen, if any :)

**Comment by johnleider** [comment]
I spoke with @nekosaur about this and for the associated PR to this issue, we are not going to include this functionality by default. With that being said, it is potentially valuable functionality, so if you could create a Feature Request we would be happy to review it.

**Comment by appurist** [comment]
@johnleider by associated PR, do you mean the one I filed (#3230) for the inconsistent input event firing?

That #3230 issue report really doesn't matter too much if the `v-model` change discussed here is applied, since the events _are_ consistent without the `:input` prop. But when you say "we are not going to include this functionality by default", does that mean not going to include _any_ of the functionality in this feature request, or just the "freeze" option part of it?

**Comment by johnleider** [comment]
Sorry, the freeze part.

**Comment by appurist** [comment]
While I too could use freeze (I have a similar vertical stepper U/I in mind), just being able to programmatically open and close the panels is a large step forward. As long as I can set the initial open/close state and default behavior, I'm reasonably happy to let the user manually override that.
